share/eds/blockstore.go

```diff
@@ -0,0 +1,195 @@
+package eds
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/filecoin-project/dagstore"
+	"github.com/filecoin-project/dagstore/shard"
+	lru "github.com/hashicorp/golang-lru"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	ipld "github.com/ipfs/go-ipld-format"
+)
+
+var _ blockstore.Blockstore = (*Blockstore)(nil)
+
+var (
+	maxCacheSize            = 100
+	ErrUnsupportedOperation = errors.New("unsupported operation")
+	ErrInvalidShardCount    = errors.New("the provided cid does not map to exactly one shard")
+)
+
+type accessorWithBlockstore struct {
+	sa *dagstore.ShardAccessor
+	bs dagstore.ReadBlockstore
+}
+
+// Blockstore implements the blockstore.Blockstore interface on an EDSStore.
+// The lru cache approach is heavily inspired by the open PR filecoin-project/dagstore/116.
+// The main differences to the implementation here are that we do not support multiple shards per key,
+// call GetSize directly on the underlying RO blockstore, and do not throw errors on Put/PutMany.
+// Also, we do not abstract away the blockstore operations.
+type Blockstore struct {
+	store *Store
+
+	// bsStripedLocks prevents simultaneous RW access to the blockstore cache for a shard. Instead of using only one
+	// lock or one lock per key, we stripe the shard keys across 256 locks
```

Comment:
what is the context behind 256 locks? might be worth including that in the comment.